zerver/lib/actions.py

```diff
@@ -862,7 +862,7 @@ def do_scrub_realm(realm: Realm, acting_user: Optional[UserProfile]=None) -> Non
                                  event_type=RealmAuditLog.REALM_SCRUBBED)
 
 def do_deactivate_user(user_profile: UserProfile,
-                       acting_user: Optional[UserProfile]=None,
+                       *, acting_user: Optional[UserProfile]=None,
```

Comment:
So, I think if we want to make this actually mandatory, we need to remove the `=None`.  We can also remove the `Optional[` in cases where we don't expect code to need to pass `None`.

Reply:
@timabbott Then I think we would have to pass acting_user=None in all the tests?. Should we record acting_user while running test cases?

Reply:
Hmm, well, we should be thoughtful.  For `do_deactivate_user` in particular, probably it doesn't make sense to make this change because we have a lot of tests that need to deactivate a user.  But for any of the `do_` functions that are only called in at most 5 places total, we should make the change I suggest.